opentracing-agent/src/test/java/io/opentracing/contrib/agent/OpenTracingHelperTest.java

```diff
@@ -16,15 +16,44 @@
  */
 package io.opentracing.contrib.agent;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
 
+import java.lang.reflect.Field;
+
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
+import io.opentracing.NoopTracerFactory;
 import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.mock.MockSpan;
 import io.opentracing.mock.MockTracer;
+import io.opentracing.util.GlobalTracer;
 
 public class OpenTracingHelperTest {
 
+    // Approach used in opentracing-util GlobalTracerTest to reset the global tracer
+    private static void _setGlobal(Tracer tracer) {
+        try {
+            Field globalTracerField = GlobalTracer.class.getDeclaredField("tracer");
+            globalTracerField.setAccessible(true);
+            globalTracerField.set(null, tracer);
+            globalTracerField.setAccessible(false);
```

Comment:
Having done QE for some time, I tend to see tests as "the first consumers of an API". Later on, they can also be used as a reference on how to properly consume your API. As such, I'm usually against this kind of approach, because you wouldn't expect your usual consumers to do that. If you have a valid reason to change an internal value of a class, your code should provide a way to do that via a getter/setter and the test should do what other consumers would do.

That said, I think that a library like Mockito could be used here, instead of using reflection to manipulate your test subjects.